File: app/src/main/java/com/faridrjb/whattocook/fragments/introslider/SlideFragment.java
package com.faridrjb.whattocook.fragments.introslider;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.faridrjb.whattocook.R;

public class SlideFragment extends Fragment {

    public static SlideFragment newSlide(int imgResId, String title, String description, @Nullable String btnNext) {
        SlideFragment fragment = new SlideFragment();
        Bundle args = new Bundle();
        args.putInt("imgResId", imgResId);
        args.putString("title", title);
        args.putString("description", description);
        args.putString("btnNext", btnNext);
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_slide, container, false);
        Button next = rootView.findViewById(R.id.btnNext102);
        if (getArguments() != null) {
            ((ImageView) rootView.findViewById(R.id.logoImage102)).setImageResource(getArguments().getInt("imgResId"));
            ((TextView) rootView.findViewById(R.id.tv103)).setText(getArguments().getString("title"));
            ((TextView) rootView.findViewById(R.id.tv104)).setText(getArguments().getString("description"));

            if (getArguments().getString("btnNext") != null) next.setText(getArguments().getString("btnNext"));
        }

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SlideLogoFragment.CallBacks activity = (SlideLogoFragment.CallBacks) getActivity();
                activity.nextClicked(v.getId());
            }
        });
        return rootView;
    }
}

File: app/src/main/java/com/faridrjb/whattocook/fragments/SlideFragment.kt
package com.faridrjb.whattocook.fragments

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import com.faridrjb.whattocook.R
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation
import androidx.viewpager.widget.ViewPager
import com.faridrjb.whattocook.databinding.FragmentSlideBinding

class SlideFragment : Fragment() {

    private var _binding: FragmentSlideBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentSlideBinding.inflate(inflater, container, false)
        val view = binding.root
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (arguments != null) {
            binding.iconIV.setImageResource(
                requireArguments().getInt("imgResId")
            )
            binding.titleTV.text =
                requireArguments().getString("title")
            binding.descTV.text =
                requireArguments().getString("description")
            if (requireArguments().getString("btnNext") != null) binding.nextBtn.text =
                requireArguments().getString("btnNext")
        }

        // Next Btn
        val rv = requireActivity().findViewById<ViewPager>(R.id.introSliderVP)
        binding.nextBtn.setOnClickListener {
            if (rv.currentItem + 1 == rv.adapter!!.count) {
                val preferences = requireContext().getSharedPreferences("First Time", AppCompatActivity.MODE_PRIVATE)
                val editor = preferences.edit()
                editor.putBoolean("First time", false)
                editor.apply()
                Navigation.findNavController(binding.root)
                    .navigate(R.id.actionIntroToMain)
            }
            rv.currentItem = rv.currentItem + 1
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        fun newSlide(
            imgResId: Int,
            title: String?,
            description: String?,
            btnNext: String?
        ): SlideFragment {
            val fragment = SlideFragment()
            val args = Bundle()
            args.putInt("imgResId", imgResId)
            args.putString("title", title)
            args.putString("description", description)
            args.putString("btnNext", btnNext)
            fragment.arguments = args
            return fragment
        }
    }
}